src/Searchs/BinarySearch.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+
+/**
+ * @title find the index position of a value in an array of sorted numbers
+ * @author Chiadikaobi https://github.com/Chiadikaobixo
+ * @dev https://en.wikipedia.org/wiki/Binary_search_algorithm
+ */
+
+/** @dev Binary search is a more faster form of searching with
+ * a timecomplexity of O(LogN), and it works on only sorted
+ * arrays. Rather than eliminating one element at a time, binary 
+ * search eliminate half of the remaining elements at a time.
+*/
+contract BinarySearch {
+    uint256 value;
+    uint256[] _nums;
+
+    // set the value to lookup in the array
+    function setValue(uint256 x) public {
+        value = x;
+    }
+    
+    // set the array of sorted numbers
+    function setArray(uint256[] memory num) public {
+        _nums = num;
+    }
+
+    // returns the index position of the value in the array
+    function printResult() public view returns(uint256 result) {
+         uint256 minimum = 0;
+         uint256 maximum = _nums.length -1;
+
+          while (minimum <= maximum) {
+            uint256 middle = (minimum + maximum)/ 2;
+            uint256 currentElement = _nums[middle];
+            if (currentElement < value) {
+              minimum = middle + 1;
+            } else if (currentElement > value) {
+              maximum = middle - 1;
+            } else {
+              result = middle;
+              // return the index position of the value in the array
+              return result;
+            }
+          }
+        // return 0 if value is not in the array
+        return 0;
+    }    
+}
```

Comment:
Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable

Reply:
@Panquesito7 kindly look into this, seems I mistakenly removed a request for @mkubdev .